listenbrainz/webserver/views/user_timeline_event_api.py

```diff
@@ -239,6 +239,57 @@ def user_feed(user_name: str):
     }})
 
 
+@user_timeline_event_api_bp.route("/user/<user_name>/feed/events/delete", methods=['OPTIONS', 'POST'])
+@crossdomain(headers="Authorization, Content-Type")
+@ratelimit()
+def delete_feed_events(user_name):
+    '''
+    Delete those events from user's feed that they created themselves.
+    Supports deletion of recommendation, notification and pin events.
+    Along with the authorization token, post one of the following, according
+    to your need.
+    {
+        "event_type": "recording_recommendation",
+        "id": int
+    }
+    {
+        "event_type": "notification",
+        "id": int
+    }
+    :param user_name: The MusicBrainz ID of the user from whose timeline events are being deleted
+    :type user_name: ``str``
+    :statuscode 200: Successful deletion
+    :statuscode 400: Bad request, check ``response['error']`` for more details.
+    :statuscode 401: Unauthorized
+    :statuscode 404: User not found
+    :statuscode 500: API Internal Server Error
+    :resheader Content-Type: *application/json*
+    '''
+    # TODO: maybe implement logging once approved
+    user = validate_auth_header()
+    if user_name != user['musicbrainz_id']:
+        raise APIUnauthorized("You don't have permissions to delete from this user's timeline.")
+
+    try:
+        event = ujson.loads(request.get_data())
+
+        if event["event_type"] in [UserTimelineEventType.RECORDING_RECOMMENDATION.value,
```

Comment:
The docstring says that pin events are supported for deletion, but this if only allows REC and NOTIFICATION events. Are pins notifications as well, or is there a case missing here?

Reply:
Pinned recordings have their own delete endpoint so I asked it to be removed from here. Ideally, we should have just one delete endpoint for all events but the pinned recs has been in prod so not sure if somebody else is using it. And having pinned recs deletion in two places may cause confusion.

Reply:
So we should fix the docstring or what?

Reply:
Oh right! I missed the docstring. Yes it should be fixed.

Reply:
yes, my bad, I will fix it in the next commit. Sorry :P